### PR TITLE
Feature/par 154 eva basic implementation for sentiment analysis

### PR DESCRIPTION
**Pull request by MarekLorenz**
- add sentiment analysis, recovery rate and similarity metrics (all three metrics are implemented in this branch)
- add sigmoid normalization function
- Add Python handler to run Python scripts

### CODE REVIEW

File: src/evaluation/service/evaluation.similarity.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def avg_jaccard(texts, n):
+    cumulative_similarity = 0
+    number_of_comparisons = 0
+    # calculate Jaccard coefficient
+    for (message1, message2) in itertools.combinations(texts, r=2):
+        grams = set([element[0] for element in ngrams(message1.split(), n)])
+        comparison = [element[0]
+                      for element in set(ngrams(message2.split(), n))]
```
Comment (tcw0) [inline src/evaluation/service/evaluation.similarity.py:17]: Indentation error

**Review by tcw0 — CHANGES_REQUESTED** [review]

File: src/evaluation/service/evaluation.python-wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * Calls the python script to calculate the recovery rate
+ * @param messages - The messages to calculate the recovery rate for.
+ * @returns Recovery rate of the agent.
+ */
+function recoveryHandler(messages: MessageDocument[]) {
+  const jsonMessages: string = JSON.stringify(messages);
+  return parseFloat(execSync(`python3 evaluation.recovery.py ${jsonMessages}`).toString());
+}
```
Comment (Fabi987) [inline src/evaluation/service/evaluation.python-wrapper.ts:22]: When using docker, I get an internal server error:
`Command failed: python3 evaluation.recovery.py [...] /bin/sh: python3: not found\n`
I also checked it: python is indeed not installed in the docker container

Are you running the development server without docker?

Comment (Fabi987) [inline src/evaluation/service/evaluation.python-wrapper.ts:22]: What happens in the case of an error in the python script? Will the error message just silently be converted to a float? (Just a question, couldn't try it out). If yes, that should be changed in my opinion.

**Review by Fabi987 — CHANGES_REQUESTED** [review]
Sorry for reviewing without being assigned haha, but I just wanted to try it out.
However, it unfortunately doesn't work with docker (see comment).
The implementation itself looks good, as far as I understand what you are doing there :sweat_smile:

File: Dockerfile [hunk 3]
```diff
@@ -1,34 +1,26 @@
...
     python3 -m spacy download en_core_web_sm
 
 # Copy the rest of your app's source code
 COPY . .
-
 # Build your TypeScript files
 RUN npm run build
-
 # The application's port
 EXPOSE 3000
```
Comment (Fabi987) [inline Dockerfile:26]: I think you should keep the Dockerfile as it is, the meta team will be probably not very happy about changes (although you have just removed lines haha)

**Review by Fabi987 — CHANGES_REQUESTED** [review]

**Review by Fabi987 — APPROVED** [review]

**Pull request merged**